blog/articles/orchestrating-cloud-events-with.md

```diff
@@ -0,0 +1,194 @@
+
+![Operate UI](https://github.com/salaboy/orchestrating-cloud-events/blob/master/imgs/tickets-v1-operate-2.png?raw=true)
+
+## Decorate / Enhance
+
+Now that you have the power of a workflow engine, you can leverage some of its features to decorate or enhance your applications. An ehnchanced version of the workflow model could look like this:
+
+![Workflow model v2](https://github.com/salaboy/orchestrating-cloud-events/blob/master/imgs/tickets-v2.png?raw=true)
+
+Now when the customer reaches the **Payment Sent** state, meaning that they have already made a reservation, the model wait for the **Payment Sent** CloudEvent to arrive.
+
+![timers](https://github.com/salaboy/orchestrating-cloud-events/blob/master/imgs/tickets-v2-timers.png?raw=true)
+
+As you can see above, two timer boundary events have been associated with the **Payment Sent** state. The bottom one, with dashed lines, is a non-interrupting event, meaning that it will not disrupt the actual flow, it will be just triggered in a fire-and-forget fashion. This result in the Payment Reminder being sent and, because it can be configured as a recurring timer, it will trigger every X amount of time (for the presentation this was set to 10 seconds).
+
+![Websockets push notification](https://github.com/salaboy/orchestrating-cloud-events/blob/master/imgs/tickets-push-notification-websocket.png?raw=true)
+
+The above image shows the notification on the frontend, which was generated by the workflow engine, routed to the website backend and forwarded using WebSockets to the website frontend in the customer browser.
+
+The top one, with solid lines, is an interrupting event, meaning that it will cause the normal flow of the model to be discarded and continue only to the steps defined from the timer, in this case, **Reservation Time Out**. For this scenario, the interrupting timer is set to be triggered after X amount (for the presentation this was set to 2 minutes) of time after the reservation of the tickets was made. After the reservation time out the website redirects the customer back to the beginning.
+
+For both timers, you get full traceability (e.g. when and how oftenthey were triggered). You can see these details graphically in **Operate**:
+
+![Workflow model v2 in Operate](https://github.com/salaboy/orchestrating-cloud-events/blob/master/imgs/tickets-v2-timers-operate.png?raw=true)
+
+You can see that the non-interrupting timer was triggered several times, generating payment reminders in the frontend. Even if we don’t implement the frontend notifications, you can use these timers to measure and classify how much time it actually takes your customer to make the payment. This can be really useful to classify different categories, for example, how much time it takes, on average, for people to provide payment details.
```

Comment:
```suggestion
You can see that the non-interrupting timer was triggered several times, generating payment reminders in the frontend. Even if we don’t implement the frontend notifications, you can use these timers to measure and classify how much time it actually takes your customer to make the payment. This can be really useful to classify different categories; for example, how much time it takes, on average, for people to provide payment details.
```